File: player/ultraPlayer/AVisibleAddon.cpp
#include "AVisibleAddon.h"

AVisibleAddon::AVisibleAddon( UINT id, AddonType::Type type, VisibleLayout* const layout ):AAddon(id, type)
{
	_layout = new VisibleLayout(*layout);
}
AVisibleAddon::~AVisibleAddon(void)
{
	CAutoLock lock(&_objectLock);
	if (_layout)
	{
		delete _layout;
	}
}

VisibleLayout* AVisibleAddon::Layout()
{
	CAutoLock lock(&_objectLock);
	
	// caller should not alter/delete instance
	return _layout;
}

void AVisibleAddon::Layout( VisibleLayout* newLayout )
{
	CAutoLock lock(&_objectLock);

	if (_layout)
	{
		delete _layout;
	}
	_layout = new VisibleLayout(*newLayout);
}

File: controller/Model/Addon/AVisibleAddon.cs
using System;
using System.Collections.Generic;
using System.Text;
using UltraPlayerController.Model.Enumerations;

namespace UltraPlayerController.Model.Addon
{
    public abstract class AVisibleAddon: AAddon
    {
        protected VisibleLayout _layout;

        public VisibleLayout Layout
        {
            get { return _layout; }
            set { _layout = value; }
        }

        public AVisibleAddon(uint id, AddonType type, VisibleLayout layout):base(id, type)
        {
            _layout = layout;
        }
    }
}
